### PR TITLE
Revert "Drop python 3.6 support"

### PR DESCRIPTION
**Pull request by balopat**
Reverts quantumlib/Cirq#3574.

Rationale: we want to ensure that Cirq works on colab which is stuck at the moment at Python 3.6.

### CODE REVIEW

**Comment by mpharrigan** [comment]
What's the ETA on colab getting a more modern Python version? Python 3.9 is out now; we should be testing against it. And now our test matrix is getting quite large

**Comment by mpharrigan** [comment]
We should follow https://numpy.org/neps/nep-0029-deprecation_policy.html 

According to that, we should be Python 3.7+ and numpy 1.17+

**Comment by 95-martin-orion** [comment]
> What's the ETA on colab getting a more modern Python version? Python 3.9 is out now; we should be testing against it. And now our test matrix is getting quite large

We have a `python 3.9` label for this - umbrella issue is #3772.

**Comment by balopat** [comment]
Opened https://github.com/quantumlib/Cirq/issues/3800 to track colab's progress.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -189,7 +193,7 @@ jobs:
     name: Pytest MacOS
     strategy:
       matrix:
-        python-version: [ '3.7', '3.8' ]
+        python-version: ['3.7', '3.8' ]
```
Comment (maffoo) [inline .github/workflows/ci.yml:196]: turbo nit: remove extra whitespace before final bracket.

**Review by maffoo — APPROVED** [review]
LGTM

**Pull request merged**